### ISSUE TITLE
(core) - Fix mutation operation's identity being used for filtering results

### ISSUE THREAD

**Issue by mlecoq** [FormidableLabs/urql#2225]
I'm trying to use urql to make a mutation but no result is returned 

urql version : `2.1.1`

**Steps to reproduce**

I have made a small example to reproduce the issue here : https://github.com/mlecoq/urql-issue 

It is a simple login form that calls my login mutation to log in

**Expected behavior**

I expect to have a result after calling it 

**Actual behavior**

I see with flipper that the network call is made but the promise never returns

**Comment by kaleabmelkie** [comment]
I got the same issue after upgrading @urql/core to 2.4.0 in a Svelte app too. The following promise does not resolve or reject, although the request was actually sent and was completed (I saw that in the dev tools Network tab).

```ts
const { data, error } = await client
    .mutation(LoginDocument, { input })
    .toPromise()
```

Also, I've noticed this happens only when used with either `cacheExchange` or `authExchange` for some reason. Removing both makes the promise resolve (I use basically every other officially supported exchange in my setup, but none of them cause this issue).

**Comment by JoviDeCroock** [comment]
Hey. we've identified the issue and will get a fix out asap

**Comment by mlecoq** [comment]
Thanks a lot for your speed !

**Comment by kitten** [comment]
No worries :+1: This should be resolved in [`@urql/core@2.4.1`](https://github.com/FormidableLabs/urql/releases/tag/%40urql%2Fcore%402.4.1). Sorry for the regression 😄
